src/teensy_sketch/teensy_sketch.ino

```diff
@@ -32,54 +34,98 @@
       * CS     - as defined in radioChipSelect field
     - Piezo buzzer
       * +ve - pin 16
-      
+
   Created 28 August 2016
   By Jamie Sanson
-  
+
   Modified 28 August 2016
   By Jamie Sanson
-  
+
 */
 
 // region includes
 #include <i2c_t3.h>
+
 #include <ESP8266.h>
 #include <Data_module.h>
+#include <Sensor_helper.h>
 #include <SPI.h>
 // end region
 
+// region pin definitions
+#define radioIntPin 0
+#define gyroIntPin 1
+#define buzzerPin 3
+#define gyroChipSelect 4
+#define radioChipSelect 5
+#define sdChipSelect 6
+// end region
+
 // region macro definitions
 #define espSerial Serial2
+#define gpsSerial Serial3
+#define gpsSerialBaud 9600
 #define debugSerialBaud 115200
-#define espSerialBaud 9600
+#define espSerialBaud 115200
 #define initFileName "init/init"
 #define dataFileName "data/data"
 #define initExtension "txt"
 #define dataExtension "csv"
 // end region
 
-// region pin definitions
-int8_t gyroChipSelect = 4;
-int8_t radioChipSelect = 5;
-int8_t radioIntPin = 15;
-int8_t sdChipSelect = 6;
+// region IMU configuration
+uint8_t OSR = ADC_8192;
+int8_t Ascale = AFS_16G;
+int8_t Gscale = GFS_250DPS;
+int8_t Mscale = MFS_16BITS;  // Choose either 14-bit or 16-bit magnetometer resolution
+int8_t Mmode = 0x02;         // 2 for 8Hz, 6 for 100Hz continuous
+uint16_t Pcal[8];         // calibration constants from MS5637 PROM registers
+unsigned char nCRC;       // calculated check sum to ensure PROM integrity
+uint32_t D1 = 0, D2 = 0;  // raw MS5637 pressure and temperature data
+double dT, OFFSET, SENS, T2, OFFSET2, SENS2;  // First order and second order corrections for raw S5637 temperature and pressure data
+
+int16_t accelCount[3];  // Stores the 16-bit signed accelerometer sensor output
+int16_t gyroCount[3];   // Stores the 16-bit signed gyro sensor output
+int16_t magCount[3];    // Stores the 16-bit signed magnetometer sensor output
+
+double Temperature, Pressure; // stores MS5637 pressures sensor pressure and temperature
+float altitudeOffset;
+
+float aRes, gRes, mRes;
+float ax,ay,az, gx, gy, gz, mx, my, mz;
+
+float gyroBias[3] = {0, 0, 0}, accelBias[3] = {0, 0, 0};
+float magCalibration[3] = {0, 0, 0}, magbias[3] = {0, 0, 0};
+
+int altitudeCount = 0;
+float altitudeBuffer = 0;
 // end region
 
 // region flags
-boolean serialDebugMode = true;
+boolean serialDebugMode = false;
+boolean initialiseOK = true;
+boolean altInit = false;
 // end region
 
 // region library instantiation
-ESP8266 wifi(espSerial);
+//ESP8266 wifi(espSerial);
```

Comment:
Should commented out code not be removed? It's cleaner and you can ways use git history to see it if you need it at a later date.